src/utils/calculations.ts

```diff
@@ -32,3 +32,22 @@ export function calculateUnderlyingAmount(props: CalculateUnderlyingAmountProps)
   const ONE_UNIT = toBN('10').pow(underlyingTokenDecimals);
   return amount.times(sharePrice).times(ONE_UNIT).toFixed(0);
 }
+
+export function computeSummaryData(labs: Pick<GeneralLabView, 'apyData' | 'DEPOSIT'>[]): SummaryData {
+  const { totalDeposits, totalEarnings } = labs.reduce(
+    ({ totalDeposits, totalEarnings }, { DEPOSIT: { userDepositedUsdc }, apyData }) => ({
+      totalDeposits: totalDeposits.plus(userDepositedUsdc),
+      totalEarnings: totalEarnings.plus(toBN(userDepositedUsdc).times(apyData)),
+    }),
+    {
+      totalDeposits: new BigNumber(0),
+      totalEarnings: new BigNumber(0),
+    }
+  );
+
+  return {
+    totalDeposits: totalDeposits.toString(),
+    totalEarnings: totalEarnings.toString(),
+    estYearlyYield: totalDeposits.isZero() ? '0' : totalEarnings.div(totalDeposits).toString(),
```

Comment:
should we fix the decimals here?

Reply:
In `src/client/routes/Labs/index.tsx` it formats correctly, see the screenshot in the PR description

```typescript
          { header: t('dashboard.holdings'), Component: <Amount value={totalDeposits} input="usdc" /> },
          { header: 'Earnings', Component: <Amount value={totalEarnings} input="usdc" /> },
          { header: 'Est. Yearly Yield', content: `${formatApy(estYearlyYield)}` },
```

Reply:
Oh yes, formatApy does the trick sorry, approved!